src/pudl_archiver/archivers/phmsagas.py

```diff
@@ -0,0 +1,51 @@
+"""Download PHMSHA data."""
+import re
+import typing
+from datetime import date
+from pathlib import Path
+
+from pudl_archiver.archivers.classes import (
+    AbstractDatasetArchiver,
+    ArchiveAwaitable,
+    ResourceInfo,
+)
+
+BASE_URL = "https://www.phmsa.dot.gov/data-and-statistics/pipeline/gas-distribution-gas-gathering-gas-transmission-hazardous-liquids"
+current_year = date.today().year
+
+
+class PhmsaGasArchiver(AbstractDatasetArchiver):
+    """PHMSA Gas Annual Reports archiver."""
+
+    name = "phmsagas"
+
+    async def get_resources(self) -> ArchiveAwaitable:
+        """Download PHMSA gas resources."""
+        link_pattern = re.compile(r"annual[-|_](\S+).zip")
+
+        for link in await self.get_hyperlinks(BASE_URL, link_pattern):
+            yield self.get_zip_resource(link, link_pattern.search(link))
+
+    async def get_zip_resource(
+        self, link: str, match: typing.Match
+    ) -> tuple[Path, dict]:
+        """Download zip file."""
+        url = f"https://www.phmsa.dot.gov/{link}"
+        file = str(match.group(1)).replace("-", "_")  # Get file name
+
+        # Set dataset partition
```

Comment:
Could you put an example of the filename pattern that we're matching in the docstring here to make the parsing below more readable?
```py
"""Download zip file.

Filenames generally look like: annual_{dataset}_{start_year}_present.zip

For example: annual_underground_natural_gas_storage_2017_present.zip
"""
```